arbor/gpu_context.hpp

```diff
@@ -0,0 +1,57 @@
+#include <memory>
+
+#ifdef ARB_HAVE_GPU
+#include <cuda.h>
+#include <cuda_runtime.h>
+#endif
+
+namespace arb {
+
+#ifndef ARB_HAVE_GPU
+struct gpu_context {
+    bool has_gpu_;
+    size_t attributes_;
+
+    gpu_context(): has_gpu_(false), attributes_(0) {}
+};
+
+#else
+
+enum gpu_flags {
+    has_concurrent_managed_access = 0,
+    has_atomic_double = 1
+};
```

Comment:
Rather than have the bit positions in the flags, I think it's simpler (and more traditional) to store the mask values in the flags, i.e. (1, 2, 4, 8 etc.).